src/modules/map/components/openlayers/OpenLayersVectorLayer.vue

```diff
@@ -46,16 +57,56 @@ export default {
                 style: this.styleUrl,
             },
         })
-        if (this.excludeSource) {
-            // we load the style on the side in order to be able to filter out some source
-            axios.get(this.styleUrl).then((response) => {
-                const vectorStyle = response.data
-                vectorStyle.layers = vectorStyle.layers.filter(
-                    (layer) => layer.source !== this.excludeSource
-                )
-                this.layer.maplibreMap.setStyle(vectorStyle)
-            })
-        }
+        this.setMapLibreStyle(this.styleUrl)
+    },
+    methods: {
+        setMapLibreStyle(styleUrl) {
+            if (this.excludeSource) {
+                // we load the style on the side in order to be able to filter out some source
+                axios.get(styleUrl).then((response) => {
+                    const vectorStyle = response.data
+                    vectorStyle.layers = vectorStyle.layers.filter(
+                        (layer) => layer.source !== this.excludeSource
+                    )
+                    this.mapLibreInstance.setStyle(vectorStyle)
+                })
+            } else if (this.layerId === VECTOR_TILES_IMAGERY_STYLE_ID) {
+                // special case here, as the imagery is only over Switzerland (for now)
+                // we inject a fair-use WMTS that covers the globe under our aerial images
+                axios.get(styleUrl).then((response) => {
+                    const vectorStyle = response.data
+                    // settings bounds for swissimage, otherwise it covers the whole world with white tiles
+                    vectorStyle.sources.swissimage_wmts.bounds = [
+                        ...proj4(CoordinateSystems.LV95.epsg, CoordinateSystems.WGS84.epsg, [
+                            TILEGRID_EXTENT[0] + 50000,
+                            TILEGRID_EXTENT[1],
+                        ]),
+                        ...proj4(CoordinateSystems.LV95.epsg, CoordinateSystems.WGS84.epsg, [
+                            TILEGRID_EXTENT[2] - 20000,
+                            TILEGRID_EXTENT[3],
```

Comment:
Why +50000 and -20000 ?

Reply:
that was arbitrary in order to reduce the bouding box of this layer, to avoid empty tiles being requested, but it didn't work well with bigger zoom level as you've seen